api/hasura/cron/epochs.ts

```diff
@@ -256,7 +301,37 @@ export async function notifyEpochEnd({
       ${usersHodlingGive.join(', ')}
     `;
 
-      if (circle.discord_webhook)
+      if (isFeatureEnabled('discord') && circle.discord_webhook) {
+        const { discord_channel_id: channelId, discord_role_id: roleId } =
+          circle?.discord_circle || {};
+
+        if (!channelId || !roleId) {
+          return null;
+        }
+
+        await sendSocialMessage({
+          message,
+          circleId: circle.id,
+          channels: {
+            discordBot: {
+              type: 'end' as const,
+              channelId,
+              roleId,
+              epochName: `Epoch ${epoch.number}`,
+              circleName: `${circle.organization?.name}/${circle.name}`,
+              endTime: epoch.end_date,
+              giveCount: epoch.token_gifts.reduce(
+                (total, { tokens }) => tokens + total,
+                0
+              ),
```

Comment:
@topocount is this correct? Assuming the `token_gifts` are the give count

Reply:
Yes, albeit an incomplete event for user opting out and the user added/removed from circle